WcaOnRails/app/webpacker/components/CompetitionsOverview/CompetitionsFilters.js

```diff
@@ -0,0 +1,394 @@
+            active={filterState.selectedYear === 'all_years'}
+          >
+            {I18n.t('competitions.index.all_years')}
+          </Dropdown.Item>
+          {PAST_YEARS_WITH_COMPETITIONS.map((year) => (
+            <Dropdown.Item
+              key={`past_select_${year}`}
+              onClick={() => dispatchFilter({ timeOrder: 'past', selectedYear: year })}
+              active={filterState.selectedYear === year}
+            >
+              {year}
+            </Dropdown.Item>
+          ))}
+        </Dropdown.Menu>
+      </Dropdown>
+    </Button>
+  );
+}
+
+function CustomDateSelector({ filterState, dispatchFilter }) {
+  const customTimeSelectionButton = (
+    <Button
+      primary
+      type="button"
+      name="state"
+      id="custom"
+      value="custom"
+      onClick={() => dispatchFilter({ timeOrder: 'custom' })}
+      active={filterState.timeOrder === 'custom'}
+    >
+      <span className="caption">{I18n.t('competitions.index.custom')}</span>
+    </Button>
+  );
+
+  return (
+    <Popup
+      on="click"
+      position="bottom center"
+      pinned
+      trigger={customTimeSelectionButton}
+    >
+      <List>
+        <List.Item>
+          <DatePicker
+            name="start-date"
+            showIcon
+            placeholderText={I18n.t('competitions.index.from_date')}
+            selected={filterState.customStartDate}
+            onChange={(date) => dispatchFilter({ customStartDate: date })}
+            maxDate={filterState.customEndDate}
+          />
+        </List.Item>
+        <List.Item>
+          <DatePicker
+            name="end-date"
+            showIcon
+            placeholderText={I18n.t('competitions.index.to_date')}
+            selected={filterState.customEndDate}
+            onChange={(date) => dispatchFilter({ customEndDate: date })}
+            minDate={filterState.customStartDate}
+          />
+        </List.Item>
+      </List>
+    </Popup>
+  );
+}
+
+function CompDisplayCheckboxes({
+  shouldIncludeCancelled,
+  dispatchFilter,
+  shouldShowRegStatus,
+  setShouldShowRegStatus,
+  displayMode,
+}) {
+  return (
+    <>
+      <div id="cancelled" className="cancel-selector">
+        <Form.Checkbox
+          label={I18n.t('competitions.index.show_cancelled')}
+          name="show_cancelled"
+          id="show_cancelled"
+          onChange={() => dispatchFilter(
+            { shouldIncludeCancelled: !shouldIncludeCancelled },
+          )}
+        />
+      </div>
+
+      {
+        displayMode === 'list'
+        && (
+          <div id="registration-status" className="registration-status-selector">
+            <Form.Checkbox
+              label={I18n.t('competitions.index.show_registration_status')}
+              name="show_registration_status"
+              id="show_registration_status"
+              onChange={() => setShouldShowRegStatus(!shouldShowRegStatus)}
+            />
```

Comment:
If this is on and I switch to map view then back to list view, it's off. Keeping it on probably makes the most sense. I can't see why it's being reset though - might be fixed by explicitly passing `checked={shouldShowRegStatus}`?

Reply:
This actually makes the display out of sync with the actually value of `shouldShowRegStatus`. But yes it's easily fixed by your suggestion